### ISSUE TITLE
[Question] PPO.learn() takes a lot longer than env.step() * [fixed step count]

### ISSUE THREAD

**Issue by SafetyMary** [DLR-RM/stable-baselines3#1695]
### ❓ Question

I have build a custom env in gymnasium and trying to train a PPO agent.

- Each "game" is expected to last **350 steps or less**
- Running ```env.step()``` 350 times will take around **14 sec**
- Running ```model = PPO('MlpPolicy', env)``` and ```model.learn(total_timesteps=int(1))``` takes around **86 sec**

It is normal to have such large difference between running the expected amount of steps in my env and learning one timestep?

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [X] If code there is, it is [minimal and working](https://github.com/DLR-RM/stable-baselines3/issues/982#issuecomment-1197044014)
- [X] If code there is, it is formatted using the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

**Comment by araffin** [comment]
> It is normal to have such large difference between running the expected amount of steps in my env and learning one timestep?

yes, because the way PPO work is that is collects `n_steps` and do `n_epochs` times gradient updates using the specified `batchsize`.
You need to update those variables if you really want to do a single training step.

Partial duplicate of https://github.com/DLR-RM/stable-baselines3/issues/1150#issuecomment-1302119777

**Comment by SafetyMary** [comment]
Given my case, what can i do to speed up training? I initially plan to do 1mil total_timesteps within 24-48 hours but that target seems quite unachievable for now. A few options in my mind,

1. Try to vectorize my env (I can deploy multiple GPUs)
    - My env is based on a shallow tf neural network output, not sure how feasible is vectorizing such things.
    - Only 14 sec out of the 86 sec update time is due to env. Seems like the bottle neck is PPO update, so not sure if speeding up my env will bring significant benefit
2. Use GPU for PPO training
   - Initial test shows it actually slows one timestep update to 97 sec

Appreciate if you could leave some pointers for me.
